### PR TITLE
Update Spree::Api::OrdersController#update to use OrderContents

### PR DESCRIPTION
**Pull request by jordan-brough**
Pull in commit from spree/spree master that updated Spree::Api::OrdersController#update to use OrderContents.

And revert previous commit from our fork that no longer applies.

### CODE REVIEW

File: api/app/controllers/spree/api/orders_controller.rb [hunk 1]
```diff
@@ -53,17 +53,12 @@ def show
...
-        if @order.update_attributes(params_without_line_items)
 
-          deal_with_line_items if params[:order][:line_items]
-
-          @order.line_items.reload
-          @order.update!
+        if @order.contents.update_cart(order_params)
+          user_id = params[:order][:user_id]
+          if current_api_user.has_spree_role?('admin') && user_id
+            @order.associate_user!(Spree.user_class.find(user_id))
```
Comment (athal7) [inline api/app/controllers/spree/api/orders_controller.rb]: can we use contents here?

Comment (athal7) [inline api/app/controllers/spree/api/orders_controller.rb]: maybe also a check (either here or in the method in contents) to make sure that we are actually changing the user could be useful as well so we don't execute unnecessary order updates through the associate_user callbacks.

Comment (jordan-brough) [inline api/app/controllers/spree/api/orders_controller.rb]: oops, I actually had intended to leave this out.  i will remove it, thanks!  (this feature isn't in 2-2-stable and there were some other commits that would be required to make it actually work iirc.  and it doesn't seem like a good feature for this endpoint anyway)

Comment (jordan-brough) [inline api/app/controllers/spree/api/orders_controller.rb]: @athal7 removed.

**Comment by athal7** [comment]
@jordan-brough i think that line may have value for an admin (in the order admin) to associate an order to a different user. does that behavior still work?

**Comment by jordan-brough** [comment]
well it never worked in spree 2.2, it was added sometime after that.  i assume we'd probably want to keep it in master when we port our changes there but it doesn't seem like a feature worth backporting to 2.2 imo.

**Comment by athal7** [comment]
i'm not sure i understand. is this introducing a regression in behavior for our store?

**Comment by jordan-brough** [comment]
nope it's not introducing any regression.  neither our 2-2-dev branch nor any spree/spree 2-2 branch supported admins modifying user_id through Spree::Api::OrdersController#update.  That feature was added [here](https://github.com/spree/spree/commit/85eddaf125e37cf2ae46db536a17f24fb3c72fa5) in Spree 2.3 apparently.  When I cherry-picked the commit that was relevant here I inadvertently picked up that code.

**Comment by athal7** [comment]
ah thanks i misunderstood.

**Comment by athal7** [comment]
looks good

**Comment by jordan-brough** [comment]
Natasha tested this on my dev box.
Waiting on Jared for benchmark data -- https://bonobos.atlassian.net/browse/DEVOPS-409.

**Comment by athal7** [comment]
just took another look, it seems that this now allows any attributes under the line items key, where previously it had whitelisted certain attributes. does it make sense to add the param whitelisting [here](https://github.com/bonobos/spree/pull/197/files#diff-173268f0a85c59ef13e0753726e16ba8R94) ?

**Comment by jordan-brough** [comment]
@athal7 per our chat it looks like we're ok on the whitelisting.

regarding the 'special line item handling' we looked at briefly in person, it looks like we're ok there also.  (for the record -- [this first change](https://github.com/spree/spree/commit/379a12275e21167c11582b1abf8052b7453973ad#diff-173268f0a85c59ef13e0753726e16ba8L48) was later effectively undone by [this later change](https://github.com/spree/spree/commit/379a12275e21167c11582b1abf8052b7453973ad#diff-173268f0a85c59ef13e0753726e16ba8L48)).

**Comment by athal7** [comment]
yep looks great!

**Comment by athal7** [comment]
@jordan-brough what's the latest on performance here?

**Pull request closed without merging**